### ISSUE TITLE
TaxInclusiveAmt missing from SalesItemLineDetail

### ISSUE THREAD

**Issue by esb** [ruckus/quickbooks-ruby#346]
This option appears to allow the price of a sales line item to be expressed as a tax inclusive value.

By law in Australia, all prices are quoted as tax inclusive, but without this option it seems to be necessary to create invoices by first removing the tax from all line items, which is a bit annoying.

Can the TaxInclusiveAmt be added to the model?

**Comment by minimul** [comment]
I thought we had everything needed for Tax Inclusive non-US invoices:
See:
https://github.com/ruckus/quickbooks-ruby/pull/171
https://github.com/ruckus/quickbooks-ruby/pull/196
https://github.com/ruckus/quickbooks-ruby/issues/268

But if that model value is needed can you put a PR? Reference previous model addition PRs
for examples.

**Comment by esb** [comment]
The global_tax_calculation only affects the display of the invoice. It controls whether you want the price of the line items displayed including or excluding tax.

The second part of the problem is to specify whether the prices on the line items include or exclude tax when creating an invoice. By default QuickBooks assumes all prices exclude tax. The problem with inclusive prices used outside the US is that for some values it's not possible to calculate a net value that will display the original inclusive price without a rounding error. This is very unfortunate because it means that sometimes its impossible to create an invoice which exactly matches the original value of the purchase.

I'll work on the updates to the code and see if I can get it working properly.

**Comment by esb** [comment]
Mmmm...I've been able to enable this option, but it appears to do absolutely nothing. No amount of fiddling around can get me an invoice populated with the correct line item values.

I did find this question - https://intuitdeveloper.lc.intuit.com/questions/1209551?_ans=1&_pos=3&_src=sapi&_tot=6&from=sapi&tags=

The response from Intuit is quite sad.

"In the linedetail, you will have to send the tax excluded amount(your code needs to do that) and tax to be added separately even if you mention taxinclusive in your request.

The api's business logic is not smart enough to handle tax inclusive amounts rigth (sic) now."

Wow. So they add the TaxInclusiveAmt attribute which should deal with this problem, but then fail to implement any logic in the API to actually implement it. 

It's very worrying that a company providing accounting software is unable to implement the code to support tax inclusive prices.
